### PR TITLE
Add /api/items/:itemId/endrent, extract DB update logic in service to model

### PR DESCRIPTION
**Pull request by synzen**
- Added `/api/items/:itemId/endrent` and its respective controller and service
- Moved the database logic in `services/items` as a method of Item in `/models/Item`
- Updated tests

let me know what you guys think about this change in logic

(i seem to have a tendency to constantly refactor lol)

### CODE REVIEW

File: server/models/Item.js [hunk 1]
```diff
@@ -11,5 +11,34 @@ const ItemSchema = new Schema({
...
+ * Stop renting an item
+ */
+async function endRent() {
+  this.set({
+    availability: false,
+    borrower: undefined,
+  });
+  await this.save();
+  return this;
+};
```
Comment (MicLi01) [inline server/models/Item.js]: availability here should be set to true for ending rent.

Comment (synzen) [reply server/models/Item.js]: good catch, fixed ~~

**Comment by synzen** [comment]
> 
> 
> availability should be true for ending rent. It would probably be a good idea to save the transaction so a user can view their past transactions.

that's a good point. for that I'm thinking of a transaction model

| Column  | Type | Purpose |
| ------------- | ------------- | -- |
| id  | ObjectId  | |
| itemId  | ObjectId | |
| borrower | ObjectId | |
| start | Date | Start of trans |
| end | Date | End of trans |
| token? | String | Maybe the token thing yu feng was talking about? this could probably be the `id` |


this can remove the `availability` since that can be derived from the start/end date, and the `borrower` as well

**Comment by Akasora39** [comment]
> > availability should be true for ending rent. It would probably be a good idea to save the transaction so a user can view their past transactions.
> 
> fixed the availability, thanks
> 
> that's a good point. for that I'm thinking of a transaction model
> 
> Column	Type	Purpose
> _id	ObjectId	Built in I'd
> itemId	ObjectId	
> borrower	ObjectId	
> start	Date	Start of trans
> end	Date	End of trans
> token?	String	Maybe the token thing yu feng was talking about? this could probably be the `_id` tbh
> this can remove the `availability` from the `Item` since that can be derived from the transaction start/end dates, and the `borrower` as well

We were discussing 

> > availability should be true for ending rent. It would probably be a good idea to save the transaction so a user can view their past transactions.
> 
> fixed the availability, thanks
> 
> that's a good point. for that I'm thinking of a transaction model
> 
> Column	Type	Purpose
> _id	ObjectId	Built in I'd
> itemId	ObjectId	
> borrower	ObjectId	
> start	Date	Start of trans
> end	Date	End of trans
> token?	String	Maybe the token thing yu feng was talking about? this could probably be the `_id` tbh
> this can remove the `availability` from the `Item` since that can be derived from the transaction start/end dates, and the `borrower` as well

We had some further discussion and the token may not be necessary as we can essentially get all the information we need from the itemId and borrower.

**Comment by yliu0715** [comment]
> > availability should be true for ending rent. It would probably be a good idea to save the transaction so a user can view their past transactions.
> 
> fixed the availability, thanks
> 
> that's a good point. for that I'm thinking of a transaction model
> 
> Column	Type	Purpose
> _id	ObjectId	Built in I'd
> itemId	ObjectId	
> borrower	ObjectId	
> start	Date	Start of trans
> end	Date	End of trans
> token?	String	Maybe the token thing yu feng was talking about? this could probably be the `_id` tbh
> this can remove the `availability` from the `Item` since that can be derived from the transaction start/end dates, and the `borrower` as well

Let's not worry about the token thing for now. It is still a bit fuzzy in my head as to how we are going to have confirmation on borrowing and renting. We can leave that for the last releases if we have time.

**Comment by synzen** [comment]
sounds guud

**Pull request merged**